respect-lib-shared/src/commonMain/kotlin/world/respect/shared/navigation/AppRoutes.kt

```diff
@@ -43,6 +46,129 @@ object Clazz : RespectAppRoute
 @Serializable
 object Report : RespectAppRoute
 
+@Serializable
+object ReportTemplateList : RespectAppRoute
+
+@Serializable
+class ReportIndictorEdit private constructor(
+    private val reportUidStr: String,
+    private val seriesIdStr: String,
+    private val indicatorJson: String? = null
+) : RespectAppRoute {
+
+    @Transient
+    val reportUid = reportUidStr.toLong()
+
+    @Transient
+    val seriesId = seriesIdStr.toInt()
+
+    @Transient
+    val indicatorData: Indicator? = indicatorJson?.let {
+        Json.decodeFromString(it)
+    }
+
+    companion object {
+        @OptIn(ExperimentalSerializationApi::class)
+        fun create(
+            reportUid: Long,
+            seriesId: Int,
+            indicator: Indicator? = null
+        ): ReportIndictorEdit {
+            val json = Json { allowTrailingComma = true }
+            return ReportIndictorEdit(
+                reportUidStr = reportUid.toString(),
+                seriesIdStr = seriesId.toString(),
+                indicatorJson = indicator?.let { json.encodeToString(Indicator.serializer(), it) }
+            )
+        }
+    }
+}
+
+@Serializable
+class ReportEditFilter private constructor(
+    private val reportUidStr: String,
+    private val seriesIdStr: String,
+    private val filterJson: String? = null
+) : RespectAppRoute {
+
+    @Transient
+    val reportUid = reportUidStr.toLong()
+
+    @Transient
+    val seriesId = seriesIdStr.toInt()
+
+    @Transient
+    val filter: ReportFilter? = filterJson?.let { Json.decodeFromString(it) }
+
+    companion object {
+        fun create(reportUid: Long, seriesId: Int): ReportEditFilter {
+            return ReportEditFilter(
+                reportUidStr = reportUid.toString(),
+                seriesIdStr = seriesId.toString()
+            )
+        }
+
+        fun create(reportUid: Long, seriesId: Int, filter: ReportFilter): ReportEditFilter {
+            return ReportEditFilter(
+                reportUidStr = reportUid.toString(),
+                seriesIdStr = seriesId.toString(),
+                filterJson = Json.encodeToString(filter)
+            )
+        }
+    }
+}
+
+@Serializable
+class ReportEdit private constructor(
+    private val reportUidStr: String,
+    private val filterJson: String? = null,
```

Comment:
filterJson, indicatorJson etc. should NOT be route parameters. They should be returned.